QKSMS/src/main/java/com/moez/QKSMS/ui/dialog/ConversationSettingsDialog.java

```diff
@@ -129,6 +133,20 @@ public boolean onPreferenceClick(Preference preference) {
                 ((MessageListActivity) getActivity()).getResultForThreadId(mThreadId);
                 getActivity().startActivityForResult(intent, RINGTONE_REQUEST_CODE);
                 break;
+
+            case SettingsFragment.PRIVATE_NOTIFICATION:
+                AlertDialog.Builder builder = new AlertDialog.Builder(mContext);
+                builder.setTitle(R.string.pref_notifications_private);
+                builder.setSingleChoiceItems(R.array.notification_private_options, mConversationPrefs.getPrivateNotificationsSetting(), (dialog, item) -> {
+                    mConversationPrefs.putString(SettingsFragment.PRIVATE_NOTIFICATION, String.valueOf(item));
```

Comment:
This use of `item` (item index) as value is the only concern I have about this enhancement. It just so happens that the values of the preference correspond to the order of the items, but I'd feel much better if there was a way to associate the items with their values array...